### PR TITLE
Move to using OutputPath for controlling output

### PR DESCRIPTION
**Pull request by jaredpar**
Based on discussions with SDK team, attempting to switch to using
`$(OutputPath)` for controlling our output directory instead of
`$(OutDir)`.  This is how the SDK recommends teams do it and need
to validate whether this will work for us or not.

The relevant SDK discussion is here 

https://github.com/dotnet/sdk/issues/1113

### CODE REVIEW

**Comment by jaredpar** [comment]
test windows_release_unit64_prtest please

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-infrastructure for review

File: src/Setup/DevDivVsix/CompilersPackage/Microsoft.CodeAnalysis.Compilers.swixproj [hunk 1]
```diff
@@ -6,15 +6,16 @@
   <PropertyGroup>
     <OutputArchitecture>neutral</OutputArchitecture>
     <OutputLocalized>false</OutputLocalized>
-    <OutputPath>$(OutDir)Insertion</OutputPath>
+    <RoslynOutputPath>$(OutputPath)</RoslynOutputPath>
+    <OutputPath>$(OutputPath)\Insertion</OutputPath>
```
Comment (jasonmalinowski) [inline src/Setup/DevDivVsix/CompilersPackage/Microsoft.CodeAnalysis.Compilers.swixproj]: This now being dependent upon itself is somewhat terrifying to me in that it's now more sensitive to ordering than it was before.

**Comment by jaredpar** [comment]
@jasonmalinowski 

> Can we have a quick bit of documentation written or a buildcop enforcement given at least to never use OutDir or something?

It's unfortunately not that simple.  We should use `$(OutDir)` necessarily in a couple of places.  

I still need to experiment with this a bit more in the context of my SDK branch.  Once I get that settled though.  I'm going to add enforcement if I can. 

As for documentation that needs to come from SDK team.  We were happy using `$(OutDir)` against pretty much every framework and MSBuild we've used for the last 5 years.  Their dependencies are driving this change (why I'm still a bit unsure).

**Comment by jasonmalinowski** [comment]
@jaredpar Once you've got something figured out that'd be appreciated. My reference wouldn't be official docs as much as a "this is what we think the code should be doing for Roslyn at least". Otherwise right now the moment somebody has a question changing something it's going to be "ask Jared".

**Comment by jaredpar** [comment]
@jasonmalinowski 

> Once you've got something figured out that'd be appreciated.

Do you mean documentation? If not can you clarify?

**Comment by jasonmalinowski** [comment]
Yes. And if "documentation" is a scary word, call it "piece of text that says what we think the repo should be doing".

**Comment by jaredpar** [comment]
It's not a scary word but it's also something I can't provide here.  The SDK team is the one who broke the ability to use OutDir, and forced us to use OutputPath.  The only context I have is the SDK bug.  Needs to be them providing the docs here.

**Comment by jasonmalinowski** [comment]
I'm OK with your documentation being "use OutputPath, I don't know why ask the SDK team".

**Comment by jaredpar** [comment]
SGTM

**Pull request merged**